### PR TITLE
Enhanced Capabilities: Enriching Vet Clinic Database for Comprehensive Analysis and Insights

### PR DESCRIPTION
**Pull request by mohashyne**
# New Features Added

**This pull request introduces exciting new features to enhance the functionality of the vet clinic database. The following additions have been made:**

 Vets Table: 
A new table named vets has been created with the following columns: 
id with an Autoincremented primary key 
name using String age: 
Integer date_of_graduation: Date

Specializations Table
To establish a numerous- to- numerous relationship between the species and stagers tables, a" join table" named specializations has been introduced.
This table manages the specialization of stagers in different species.

Visits Table Another" join table" named visits has been added to manage the numerous- to- numerous relationship between the creatures and stagers tables. also, this table keeps track of the date of the visit.

Data Insertion Data has been fitted to colonize the new tables and prisoner meaningful connections
stagers data has been added, including information about their names, periods, and scale dates.
Specialization data reflects the specialization of stagers in different species.
Visits data captures cases of creatures visiting specific stagers on certain dates.

Example Data Insertion

warhorse William Tatcher, specializing in Pokemon.
warhorse Stephanie Mendez, specializing in Digimon and Pokemon.
warhorse Jack Harkness, specializing in Digimon.
Visits similar as Agumon visiting William Tatcher on May 24th, 2020, and Pikachu visiting Maisy Smith on multiple occasions.

These new features expand the capabilities of the warhorse clinic database, enabling more detailed record- keeping and analysis. The connections between stagers, species, and creatures have been amended, furnishing a comprehensive view of the clinic's operations. The structured data insertion enhances the authenticity of the database and allows for meaningful queries and perceptivity.

**Screenshot of  New tables Added**

# Some query quest for visits  by vets and specialization as requirement in this project:

- [ ] Who was the last animal seen by William Tatcher?
- [ ] How many different animals did Stephanie Mendez see?
- [ ] List all vets and their specialties, including vets with no specialties.
- [ ] List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- [ ] What animal has the most visits to vets?
- [ ] Who was Maisy Smith's first visit?
- [ ] Details for most recent visit: animal information, vet information, and date of visit.
- [ ] How many visits were with a vet that did not specialize in that animal's species?
- [ ] What specialty should Maisy Smith consider getting? Look for the species she gets the most

query1
query2
query3
query4

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -84,4 +84,43 @@ UPDATE animals
...
+    (18, 4, '2021-04-07'),
+    (19, 3, '2019-09-29'),
+    (20, 2, '2020-10-03'),
+    (20, 2, '2020-11-04'),
+    (4, 1, '2019-01-24'),
+    (4, 1, '2019-05-15'),
+    (4, 1, '2020-02-27'),
+    (4, 1, '2020-08-03'),
+    (22, 3, '2020-05-24'),
+    (22, 1, '2021-01-11');
```
Comment (abdulhamiid) [inline data.sql]: It seems you have entered the wrong data in your `data.sql` file.

Kindly check to ensure the correctness of the `screenshots` of your query results 🛠 🙏

File: queries.sql [hunk 2]
```diff
@@ -118,4 +118,73 @@ FROM owners o
...
+SELECT s.name AS specialty_name, COUNT(*) AS num_visits
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets vt ON v.vet_id = vt.id
+JOIN specializations sp ON vt.id = sp.vet_id
+JOIN species s ON sp.species_id = s.id
+WHERE vt.name = 'Maisy Smith'
+GROUP BY s.id
+ORDER BY num_visits DESC
 LIMIT 1;
```
Comment (abdulhamiid) [inline queries.sql]: Please add the correct screenshot of the following queries to your Pull Request description. 🛠

- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @mohashyne 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Understand the different types of relationships between tables.
- ✔️ Use primary key & foreign key mechanisms for joining tables.


## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mohashyne** [comment]
All requirements have been updated

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @mohashyne,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message.
✅ Documented your work professionally.
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**